### PR TITLE
[docs] add dbt code references guide

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Brief expansion of the code references guide to include info on generating dbt code references.

## Test Plan

vercel

### CODE REVIEW

**Comment by benpankow** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22318?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#22319** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22319?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#22318** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22318?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#21755** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21755?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/content/guides/dagster/code-references.mdx [hunk 1]
```diff
@@ -58,6 +60,43 @@ A link to the asset's source in `with_source_code_references.py` will then be vi
...
+
+from dagster_dbt import (
+    DagsterDbtTranslator,
+    DagsterDbtTranslatorSettings,
+    DbtCliResource,
+    dbt_assets,
+)
+
+from dagster import AssetExecutionContext, Definitions
+from dagster._core.definitions.metadata import with_source_code_references
```
Comment (sryza) [inline docs/content/guides/dagster/code-references.mdx:78]: We shouldn't put any private imports in docs

**Review by sryza — CHANGES_REQUESTED** [review]
For consistency with how we treat the dbt facets of other features, I think this should go in the dbt reference page, instead of here.

File: docs/content/guides/dagster/code-references.mdx (hunk 1 above)
Comment (benpankow) [reply docs/content/guides/dagster/code-references.mdx:78]: Yeah, this is just for demonstration purposes until we expose these apis publicly - will do another pass once we do so

**Comment by benpankow** [comment]
Updated, moved to dbt docs w/ link from references guide

File: docs/content/integrations/dbt/reference.mdx [hunk 2]
```diff
@@ -520,6 +520,43 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
 
 Dagster also supports fetching additional metadata at dbt execution time to attach to asset materializations. For more information, see the [Customizing asset materialization metadata](#customizing-asset-materialization-metadata) section.
 
+#### Attaching code reference metadata
+
+Dagster's dbt integration can automatically attach [Code reference](/guides/dagster/code-references) metadata to the SQL files backing your dbt assets. To enable this feature, set the `enable_code_references` parameter to `True` in the <PyObject module="dagster_dbt" object="DagsterDbtTranslatorSettings" /> passed to your <PyObject module="dagster_dbt" object="DagsterDbtTranslator" />:
```
Comment (sryza) [inline docs/content/integrations/dbt/reference.mdx:525]:
```suggestion
Dagster's dbt integration can automatically attach [code reference](/guides/dagster/code-references) metadata to the SQL files backing your dbt assets. To enable this feature, set the `enable_code_references` parameter to `True` in the <PyObject module="dagster_dbt" object="DagsterDbtTranslatorSettings" /> passed to your <PyObject module="dagster_dbt" object="DagsterDbtTranslator" />:
```

File: docs/content/integrations/dbt/reference.mdx [hunk 3]
```diff
@@ -520,6 +520,43 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
...
+
+@dbt_assets(
+    manifest=manifest_path,
+    dagster_dbt_translator=dagster_dbt_translator,
+)
+def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
+    yield from dbt.cli(["build"], context=context).stream()
+
+
+# optionally, add references to the Python source with with_source_code_references
```
Comment (sryza) [inline docs/content/integrations/dbt/reference.mdx:556]: I think this is distracting here and doesn't need to be mentioned.

**Review by sryza — APPROVED** [review]
The non-dbt parts need to wait until the utility function is exposed publicly, but the section in the dbt reference could be merged now.

**Comment by benpankow** [comment]
### Merge activity

* **Jul 8, 1:45 PM PDT**: @benpankow started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/22318).

**Pull request merged**